tests/ml_tools/test_checkpointing.py

```diff
@@ -58,7 +87,7 @@ def test_random_basicqQM_save_load_ckpts(BasicQuantumModel: QuantumModel, tmp_pa
 
     def loss_fn(model: QuantumModel, data: torch.Tensor) -> tuple[torch.Tensor, dict]:
         next(cnt)
-        out = model.expectation({})
+        out = model.expectation({}).squeeze(dim=0)
```

Comment:
Would this work in all cases?

Reply:
I don't think so, just for this particular dummy loss, since we are using `torch.rand(1)` as the target. I changed so that the dimensions match, otherwise the expectation values returns a shape [1,1]. I could also reshape the target if that's clearer?

Reply:
I changed it in this [commit](https://github.com/pasqal-io/qadence/pull/477/commits/66351310c2c8891a6f198cbedad681d721f2b5ac)

Reply:
Yeah,  but why?
I guess because the tests failed, but I don' t understand why

Reply:
Tests were not failing but without squeezing they were throwing this kind of warning:
` UserWarning: Using a target size (torch.Size([1])) that is different to the input size (torch.Size([1, 1])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
    return F.mse_loss(input, target, reduction=self.reduction)`

Reply:
I just changed it to get rid of the warning